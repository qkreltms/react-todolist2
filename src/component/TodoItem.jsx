import React from "react";
import PropTypes from 'prop-types'

import { withStyles } from '@material-ui/core/styles'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography';

const styles = (theme) => ({
    paper: {
        margin: theme.spacing.unit * 2
    }
})

const TodoItem = ({ todo, deleteTodo, classes }) => {
    //onClick을 arrow function onClick시 this가 해당 태그가 됨 그러므로 arrow를 써서 상위 영역을 가리키게함
    return (
        <Grid item xs>
            <Paper className={classes.paper}>
                <Typography noWrap onClick={() => {deleteTodo(todo.id)}}>
                    {todo.todo}
                </Typography>
            </Paper>
        </Grid>
    )
};

TodoItem.propTypes = {
    todo: PropTypes.shape({
        id: PropTypes.number,
        todo: PropTypes.string,
        deleteTodo: PropTypes.func
    }),
    classes: PropTypes.object.isRequired
}

TodoItem.defaultProps = {
    todo: PropTypes.shape({
        id: 0,
        todo: '',
        deleteTodo: () => console.warn('deleteTodo is not defined')
    }),
  classes: () => console.warn('classes is not defined')
}

export default withStyles(styles)(TodoItem)
