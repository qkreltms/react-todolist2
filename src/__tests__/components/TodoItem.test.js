import React from 'react';
import TodoItem from '../../component/TodoItem';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('TodoItem', () => {
    test('renders without crashing', () => {
        const component = shallow(<TodoItem />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
      })
})
