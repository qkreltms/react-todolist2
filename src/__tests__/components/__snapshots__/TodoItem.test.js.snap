// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem renders without crashing 1`] = `
<TodoItem
  classes={
    Object {
      "paper": "TodoItem-paper-1",
    }
  }
  deleteTodo={[Function]}
  todo={
    Object {
      "id": 0,
      "msg": "",
    }
  }
/>
`;
